### ISSUE TITLE
Request to share reproducible training code

### ISSUE THREAD

**Issue by codeslake** [shallowdream204/SODA-SR#11]
Thanks @shallowdream204 for sharing the final checkpoints of SwinIR (#10)!

I have a few follow up questions though,

- Please share reproducible training code of the proposed self-training scheme.

  - I've been trying to reproduce the quantitative table of the paper (both CVPR official (CDC) and arXiv v1,version (SwinIR)). However, I cannot reproduce the numbers. Note that for SwinIR, I've been using the exact same code (main_train_sodasr.py) uploaded in the repo. Also, please note that I've also tried experimenting models with Eqs. 10 and 11.
    - I've reproduced "source-only" models (target-only rows for each column) for each CDC and SwinIR, but I could not reproduce numbers for SODA-SRed models.
    - Whenever I enable WAT (as in the last column of Table 3), the final PSNR is 30.73dB for CDC and 30.06dB for SwinIR. When I disable WAT (as in 3rd column in Table 3), I get higher PSNR, 30.94dB for CDC and 30.85dB for SwinIR, which are far lower that what you guys report in the paper.
    - Sharing final check points really helps, but I'd like to ask authors to kindly share the main contribution of the paper, the source-free self-training scheme with WAT. Can you guys please share reproducible training code?
  - About CDC, if it is hard for you guys to share the code at the moment, can you please tell where exactly Gumbel softmax is applied in the architecture, and WAT is applied?

**Comment by Renzhihan** [comment]
Hi, do you know how to implement Eqs. 10 and 11, especially the wavelet decomposition? I can't find the code on train_test_scripts/main_train_sodasr.py.
